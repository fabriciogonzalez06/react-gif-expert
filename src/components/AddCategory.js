import React, { useState } from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');
    const handledInputChange = (e) => {
        setinputValue(e.target.value);
        console.log('change input change llamado')
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length>2){

            setCategories((categories)=> [inputValue,...categories]);
            setinputValue('');
        }
    }

    return (

        <form onSubmit={handleSubmit}>
            <input
                placeholder="Ingrese una categoria y presione enter"
                onChange={handledInputChange}
                value={inputValue}
                className="form-control mb-2"
                type="text"
            />
        </form>

    )
}

AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
}

export default AddCategory;
