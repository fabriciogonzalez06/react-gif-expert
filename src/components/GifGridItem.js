import React from 'react'
import PropTypes from 'prop-types';

function GifGridItem({ id, title, url }) {
    return (
        <div className="card1 animate__animated animate__bounce" >

            <img src={url} alt={title} />
            <p> {title} </p>
          
        </div>

    )
}

GifGridItem.propTypes = {
    title:PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    
}

export default GifGridItem;
