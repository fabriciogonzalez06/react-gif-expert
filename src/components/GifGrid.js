// import React, { useEffect, useState } from 'react'
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types'
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ categoria }) => {

    //useEffect solo se ejecuta una vez, es como en las clases el didMount
    // useEffect(() => {
    //    getGifs(categoria).then(setimages);
    // }, [categoria]);

    // const [images, setimages] = useState([]);

    const { data:images, loading }= useFetchGifs(categoria);
    




    return (
        <>
            <h1> {categoria} </h1>

            {loading && <p>Cargando..</p>  }
            <div className="card-grid" >
                {
                    images.map((img) => <GifGridItem key={img.id}  {...img} />)
                }
            </div>
        </>
    )
}

GifGrid.prototype={
    categoria:PropTypes.string.isRequired
}
