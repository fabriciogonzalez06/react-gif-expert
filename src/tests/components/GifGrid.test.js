import React from 'react';
import {GifGrid }from '../../components/GifGrid';
import { shallow} from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en gifGrid', () => {

    test('Deberia de mostrase correctamente', () => {

        const categoria='Onde Punch';
        const wrapper= shallow(<GifGrid categoria={categoria} />);

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imagenes con nuestro UseFetchGifs', () => {
        
        const gifs=[{id:'ABC',url:'https://localhost/hola.png',title:'cualquier cosa'}];

        //crear mock
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });

        const categoria='Onde Punch';
        const wrapper= shallow(<GifGrid categoria={categoria} />);
        

        expect(wrapper).toMatchSnapshot();


    })
    
    
})
