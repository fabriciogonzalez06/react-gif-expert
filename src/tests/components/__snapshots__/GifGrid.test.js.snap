// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gifGrid Deberia de mostrase correctamente 1`] = `
<Fragment>
  <h1>
     
    Onde Punch
     
  </h1>
  <p>
    Cargando..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en gifGrid debe de mostrar items cuando se cargan imagenes con nuestro UseFetchGifs 1`] = `
<Fragment>
  <h1>
     
    Onde Punch
     
  </h1>
  <p>
    Cargando..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
